src/main/java/de/qabel/desktop/hockeyapp/VersionClient.java

```diff
@@ -0,0 +1,122 @@
+package de.qabel.desktop.hockeyapp;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.protocol.HTTP;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+public class VersionClient {
+
+    public static final String API_VERSIONS_NEW = "/app_versions/new";
+    public static final String API_VERSIONS_ALL = "/app_versions";
+    private final HockeyAppConfiguration config;
+    public HttpClient httpClient;
+    private List<HockeyAppVersion> versions = new LinkedList<>();
+    private HockeyAppVersion version;
+
+    public VersionClient(HockeyAppConfiguration config, HttpClient httpClient) {
+        this.httpClient = httpClient;
+        this.config = config;
+    }
+
+
+    void loadVersions() throws IOException, JSONException {
+
+        HttpGet httpGet = config.getHttpGet(API_VERSIONS_ALL);
+
+        HttpResponse response = httpClient.execute(httpGet);
+        String responseContent = EntityUtils.toString(response.getEntity());
+
+        parseVersionsResponse(responseContent);
+    }
+
+    void findAndLoadVersion(String shortVersion) throws IOException {
+        getVersions().forEach(version -> {
+            if (shortVersion.equals(version.getShortVersion())) {
+                setVersion(version);
```

Comment:
the structure of this method is weird.
```JAVA
for(HockeyAppVersion version : getVersions()) {
    if (shortVersion.equals(version.getShortVersion())) {
        setVersion(version)
        return;
    }
}
createVersion(shortVersion);
```

would be a simpler variant of this method but it's not very consistent: findAndLoad implies some returned version, but nothing is returned. Instead `this.version` is set but only used in the caller (`getVersion()`).
So the intent would imho become more obvious if you would split this into `findVersion` that either returns the found version or throws an exception and another method (maybe directly in `getVersion()`) that checks if a version is already set, calls findVersion if not and creates a new version on failure.

In addition to that `VersionClient::getVersion` method that might do remote calls but might not is pretty obfuscated. The caller can't tell if remote calls are done and `getVersion` implies not while the `Client` implies remote calls. (Never make remote calls implicit!).